src/clusterfuzz/_internal/datastore/data_handler.py

```diff
@@ -1593,45 +1593,82 @@ def get_testcase_variant(testcase_id, job_type):
 FUZZ_TARGET_UPDATE_FAIL_DELAY = 2
 
 
-@retry.wrap(
-    retries=FUZZ_TARGET_UPDATE_FAIL_RETRIES,
-    delay=FUZZ_TARGET_UPDATE_FAIL_DELAY,
-    function='datastore.data_handler.record_fuzz_target')
 def record_fuzz_target(engine_name, binary_name, job_type):
-  """Record existence of fuzz target."""
-  if not binary_name:
-    logs.log_error('Expected binary_name.')
-    return None
+  """Records exsistence of fuzz target to the DB."""
+  result = record_fuzz_targets(engine_name, [binary_name], job_type)[0]
 
   project = get_project_name(job_type)
   key_name = data_types.fuzz_target_fully_qualified_name(
       engine_name, project, binary_name)
 
-  fuzz_target = ndb.Key(data_types.FuzzTarget, key_name).get()
-  if not fuzz_target:
-    fuzz_target = data_types.FuzzTarget(
-        engine=engine_name, project=project, binary=binary_name)
-    fuzz_target.put()
-
-  job_mapping_key = data_types.fuzz_target_job_key(key_name, job_type)
-  job_mapping = ndb.Key(data_types.FuzzTargetJob, job_mapping_key).get()
-  if job_mapping:
-    job_mapping.last_run = utils.utcnow()
-  else:
-    job_mapping = data_types.FuzzTargetJob(
-        fuzz_target_name=key_name,
-        job=job_type,
-        engine=engine_name,
-        last_run=utils.utcnow())
-  job_mapping.put()
-
   logs.log(
       'Recorded use of fuzz target %s.' % key_name,
       project=project,
       engine=engine_name,
       binary_name=binary_name,
       job_type=job_type)
-  return fuzz_target
+  return result
+
+
+def get_or_create_multi(mapping, data_type):
+  """Gets or creates multiple db entities."""
+  keys = list(mapping.keys())
+  entities = ndb.get_multi([ndb.Key(data_type, key) for key in mapping])
```

Comment:
use the ndb_utils variants for these: https://github.com/google/clusterfuzz/blob/7bff389c58f58bab0c58682697671660ebfc94da/src/clusterfuzz/_internal/datastore/ndb_utils.py#L53

These handle higher batch counts than what ndb allows by default.